src/hotspot/share/runtime/deoptimization.cpp

```diff
@@ -958,10 +958,10 @@ Deoptimization::DeoptAction Deoptimization::_unloaded_action
 template<typename CacheType>
 class BoxCacheBase : public CHeapObj<mtCompiler> {
 protected:
-  static InstanceKlass* find_cache_klass(Symbol* klass_name) {
-    ResourceMark rm;
+  static InstanceKlass* find_cache_klass(Thread* thread, Symbol* klass_name) {
```

Comment:
For next time ... if `thread` is always the current thread it would be good to name it `current`.